### PR TITLE
[MAINTENANCE] Rename ValidationConfig to ValidationDefinition

### PR DESCRIPTION
**Pull request by billdirks**
- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -330,7 +330,9 @@ def _init_factories(self) -> None:
                 context=self,
             )
 
-        self._validations: ValidationFactory = ValidationFactory(store=self.validation_config_store)
+        self._validations: ValidationFactory = ValidationFactory(
```
Comment (cdkini) [inline great_expectations/data_context/data_context/abstract_data_context.py]: Should we rename this to `validation_definitions`?

File: great_expectations/data_context/types/base.py [hunk 2]
```diff
@@ -2410,7 +2410,7 @@ def __str__(self) -> str:
         return self.__repr__()
 
 
-class CheckpointValidationConfig(AbstractConfig):
+class CheckpointValidationDefinition(AbstractConfig):
```
Comment (cdkini) [inline great_expectations/data_context/types/base.py:2413]: Don't think we're actually going to use this class anymore but I'm okay with the change

File: tests/checkpoint/test_v1_checkpoint.py [hunk 3]
```diff
@@ -71,46 +71,46 @@ def in_memory_context(self) -> EphemeralDataContext:
...
             yield in_memory_context.validations.add(vc)
 
     @pytest.fixture
-    def validation_config_2(
+    def validation_definition_2(
         self, in_memory_context: EphemeralDataContext, mocker: pytest.MockFixture
     ):
         name = "my_second_validation"
-        vc = ValidationConfig(
+        vc = ValidationDefinition(
```
Comment (cdkini) [inline tests/checkpoint/test_v1_checkpoint.py:95]: Minor nit - `vc` is used as a variable in a few places maybe we change those to `vd`?

**Review by cdkini — APPROVED** [review]

File: tests/checkpoint/test_v1_checkpoint.py (hunk 3 above)
Comment (billdirks) [reply tests/checkpoint/test_v1_checkpoint.py:95]: Changed to `validation_definition`

File: great_expectations/data_context/data_context/abstract_data_context.py (hunk 1 above)
Comment (billdirks) [reply great_expectations/data_context/data_context/abstract_data_context.py]: Renamed. There is a public property called `validations` which I will leave for the time being but will update in a followup PR.

**Pull request merged**